src/player/Player.php

```diff
@@ -2806,20 +2824,31 @@ public function removeCurrentWindow() : void{
 		}
 	}
 
-	protected function addPermanentInventories(Inventory ...$inventories) : void{
-		foreach($inventories as $inventory){
-			$inventory->onOpen($this);
-			$this->permanentWindows[spl_object_id($inventory)] = $inventory;
+	/**
+	 * @param PlayerInventoryWindow[] $windows
+	 */
+	protected function addPermanentWindows(array $windows) : void{
+		foreach($windows as $window){
+			$window->onOpen();
+			$this->permanentWindows[spl_object_id($window)] = $window;
```

Comment:
This can probably now create confusion ? Before, the inventory is normally the same instance while now the window could be different with the same inventory instance.